coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1387,13 +1387,27 @@ func InsertWorkspaceResource(ctx context.Context, db database.Store, jobID uuid.
 				Valid:  true,
 			}
 		}
-		var env pqtype.NullRawMessage
-		if prAgent.Env != nil {
-			data, err := json.Marshal(prAgent.Env)
+
+		env := make(map[string]string)
+		// For now, we only support adding extra envs, not overriding
```

Comment:
We should add this comment to provider docs too so that anyone using the provider knows this beforehand.